lib/internal/Magento/Framework/Mview/View.php

```diff
@@ -224,34 +211,30 @@ public function subscribe()
     /**
      * Remove subscriptions
      *
-     * @throws \Exception
      * @return ViewInterface
+     * @throws Exception
      */
     public function unsubscribe()
     {
         if ($this->getState()->getMode() != View\StateInterface::MODE_DISABLED) {
-            try {
-                // Remove subscriptions
-                foreach ($this->getSubscriptions() as $subscriptionConfig) {
-                    /** @var \Magento\Framework\Mview\View\SubscriptionInterface $subscription */
-                    $subscriptionInstance = $this->subscriptionFactory->create(
-                        [
-                            'view' => $this,
-                            'tableName' => $subscriptionConfig['name'],
-                            'columnName' => $subscriptionConfig['column'],
-                            'subscriptionModel' => !empty($subscriptionConfig['subscriptionModel'])
-                                ? $subscriptionConfig['subscriptionModel']
-                                : SubscriptionFactory::INSTANCE_NAME,
-                        ]
-                    );
-                    $subscriptionInstance->remove();
-                }
-
-                // Update view state
-                $this->getState()->setMode(View\StateInterface::MODE_DISABLED)->save();
-            } catch (\Exception $e) {
-                throw $e;
+            // Remove subscriptions
+            foreach ($this->getSubscriptions() as $subscriptionConfig) {
+                /** @var View\SubscriptionInterface $subscription */
+                $subscriptionInstance = $this->subscriptionFactory->create(
```

Comment:
Looks like it could be replaced to ` $subscriptionInstance = $this->initSubscriptionInstance($subscriptionConfig);`

Reply:
There's inconsistency with 'subscription_model' and 'subscriptionModel', however I'll try to handle that.

Reply:
OK. According to the https://github.com/magento/magento2/blob/15e3c07f3aeb94d4147e49401d409afc3dcd93c9/lib/internal/Magento/Framework/Mview/Config/Converter.php#L89-L93 - This should be `subscription_model` not `subscriptionModel`